Jellyfin.Server.Implementations/Devices/DeviceManager.cs

```diff
@@ -0,0 +1,243 @@
+
+            if (query.DeviceId != null)
+            {
+                devices = devices.Where(device => device.DeviceId == query.DeviceId);
+            }
+
+            if (query.AccessToken != null)
+            {
+                devices = devices.Where(device => device.AccessToken == query.AccessToken);
+            }
+
+            var count = await devices.CountAsync().ConfigureAwait(false);
+
+            if (query.Skip.HasValue)
+            {
+                devices = devices.Skip(query.Skip.Value);
+            }
+
+            if (query.Limit.HasValue)
+            {
+                devices = devices.Take(query.Limit.Value);
+            }
+
+            return new QueryResult<Device>
+            {
+                Items = await devices.ToListAsync().ConfigureAwait(false),
+                StartIndex = query.Skip ?? 0,
+                TotalRecordCount = count
+            };
+        }
+
+        /// <inheritdoc />
+        public async Task<QueryResult<DeviceInfo>> GetDeviceInfos(DeviceQuery query)
+        {
+            var devices = await GetDevices(query).ConfigureAwait(false);
+
+            return new QueryResult<DeviceInfo>
+            {
+                Items = devices.Items.Select(device => ToDeviceInfo(device)).ToList(),
+                StartIndex = devices.StartIndex,
+                TotalRecordCount = devices.TotalRecordCount
+            };
+        }
+
+        /// <inheritdoc />
+        public async Task<QueryResult<DeviceInfo>> GetDevicesForUser(Guid? userId, bool? supportsSync)
+        {
+            await using var dbContext = _dbProvider.CreateContext();
+            var sessions = dbContext.Devices
+                .Include(d => d.User)
+                .AsQueryable()
+                .OrderBy(d => d.DeviceId)
+                .ThenByDescending(d => d.DateLastActivity)
+                .AsAsyncEnumerable();
+
+            if (supportsSync.HasValue)
+            {
+                sessions = sessions.Where(i => GetCapabilities(i.DeviceId).SupportsSync == supportsSync.Value);
+            }
+
+            if (userId.HasValue)
+            {
+                var user = _userManager.GetUserById(userId.Value);
+
+                sessions = sessions.Where(i => CanAccessDevice(user, i.DeviceId));
+            }
+
+            var array = await sessions.Select(device => ToDeviceInfo(device)).ToArrayAsync().ConfigureAwait(false);
+
+            return new QueryResult<DeviceInfo>(array);
+        }
+
+        /// <inheritdoc />
+        public async Task DeleteDevice(Device device)
+        {
+            await using var dbContext = _dbProvider.CreateContext();
+            dbContext.Devices.Remove(device);
+            await dbContext.SaveChangesAsync().ConfigureAwait(false);
+        }
+
+        /// <inheritdoc />
+        public bool CanAccessDevice(User user, string deviceId)
+        {
+            if (user == null)
+            {
+                throw new ArgumentException("user not found");
```

Comment:
Shouldn't it be `ArgumentNullException`?
```suggestion
                throw new ArgumentNullException(nameof(user));
```